### PR TITLE
[9.x] add search method to eloquent

### PR DESCRIPTION
**Pull request by abbasudo**
Add new `search()` method to eloquent.
Sometimes we need to search a column for a specific text . insted of using raw query with `LIKE` , i added `search` method so you can use it like this: 
```php
Post::search('title' , 'something')->get();
```

i have added a test for search and i will make changes to the document in case you accepted it.
thanks for your time and attention.

### CODE REVIEW

**Comment by driesvints** [comment]
This will conflict with Scout.

**Comment by llabbasmkhll** [comment]
@driesvints would you suggest me another method name. dose `including()` fits?

File: src/Illuminate/Database/Eloquent/Builder.php [hunk 1]
```diff
@@ -247,6 +247,20 @@ public function where($column, $operator = null, $value = null, $boolean = 'and'
...
+    /**
+     * Add a basic like clause to the query.
+     *
+     * @param  \Closure|string|array|\Illuminate\Database\Query\Expression  $column
+     * @param  mixed  $value
+     * @return $this
+     */
+    public function search($column, $value)
+    {
+        $this->where($column, 'LIKE', "%{$value}%");
```
Comment (crynobone) [inline src/Illuminate/Database/Eloquent/Builder.php:259]: Not all database support `LIKE`

**Comment by driesvints** [comment]
@llabbasmkhll I don't think we should merge this. You can add this syntax to your own app if you want.

**Comment by llabbasmkhll** [comment]
@driesvints Ok cool . thanks for reviewing.

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If possible, please consider releasing your code as a package so that the community can still take advantage of your contributions! 

If you feel absolutely certain that this code corrects a bug in the framework, please "**@**" mention me in a follow-up comment with further explanation so that GitHub will send me a notification of your response.

**Comment by SocolaDaiCa** [comment]
@llabbasmkhll how to search special character https://stackoverflow.com/questions/712580/list-of-special-characters-for-sql-like-clause

**Pull request closed without merging**